### PR TITLE
!refactor: Updates Amundsen to use new version of FlaskOIDC

### PR DESCRIPTION
**Pull request by verdan**
Signed-off-by: verdan <verdan.mahmood@gmail.com>

### Summary of Changes

This is a breaking change for anyone using Amundsen with OIDC. I have released a new version of `flaskoidc` which works with the Discovery endpoints. That means we don't need to take care of clients_secrets.json files anymore. 
People will still be able to use the `flaskoidc==0.2.3` as is, with the clients_secrets.json file if they want. 
From `1.0.0` I've dropped the changed the underlying package, and moved away from `flask-oidc` in favor of `authlib`. 

Configuration is documented here: https://github.com/verdan/flaskoidc#readme
There will be a blog post on what's new in `flaskoidc` soon. 

The new config will look something like this:
### For Frontend
Screenshot 2021-06-18 at 10 54 24

### For Metadata and Search
Screenshot 2021-06-18 at 10 55 01

### Documentation

Updated the documentation to mention both the old and new versions and how to configure both. 

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by mgorsk1** [comment]
since it's a breaking change anyways maybe it's a good opportunity to unify env var names (APP_WRAPPER <> FLASK_APP_MODULE_NAME and APP_WRAPPER_CLASS <> FLASK_APP_CLASS_NAME )?

**Comment by verdan** [comment]
the scope will be changed for this PR then, as it does not change any code for Amundsen OSS. 
For now, only the people using `flaskoidc` will be affected, and then anyone using customisations will be affected. thoughts?

**Comment by mgorsk1** [comment]
I think we could just fallback new env variable to old one (if present) and log deprecation warning, while updating the docs to suggest the new name. It would provide safe cushion for customization users.

**Comment by verdan** [comment]
@mgorsk1 done

File: frontend/amundsen_application/__init__.py [hunk 1]
```diff
@@ -21,15 +22,29 @@
...
 
+# For customized flask use below arguments to override.
+FLASK_APP_MODULE_NAME = os.getenv('FLASK_APP_MODULE_NAME') or os.getenv('APP_WRAPPER')
+FLASK_APP_CLASS_NAME = os.getenv('FLASK_APP_CLASS_NAME') or os.getenv('APP_WRAPPER_CLASS')
+FLASK_APP_KWARGS_DICT_STR = os.getenv('FLASK_APP_KWARGS_DICT') or os.getenv('APP_WRAPPER_ARGS')
 
-app_wrapper_class = Flask
+# Deprecation Warnings
+warnings.simplefilter('always', DeprecationWarning)
+if os.getenv('APP_WRAPPER') and os.getenv('APP_WRAPPER_CLASS'):
```
Comment (mgorsk1) [inline frontend/amundsen_application/__init__.py:32]: whould having `or` instead of `and` hurt us?

Comment (mgorsk1) [inline frontend/amundsen_application/__init__.py:32]: whould having `or` instead of `and` hurt us?

**Review by mgorsk1 — COMMENTED** [review]
one nit

**Review by mgorsk1 — APPROVED** [review]

File: docs/authentication/oidc.md [hunk 2]
```diff
@@ -7,29 +7,32 @@ the views. It also accepts headers on each request if available in order to vali
...
 
+
 ## Amundsen Configuration
 
 Once you have `flaskoidc` installed and configured for each microservice, please set the following environment variables:
 
-- amundsenfrontendlibrary:
+- amundsenfrontendlibrary (`amundsen/frontend`):
 ```bash
     APP_WRAPPER: flaskoidc
```
Comment (mgorsk1) [inline docs/authentication/oidc.md:25]: old vars referenced

**Review by mgorsk1 — APPROVED** [review]

**Review by dorianj — APPROVED** [review]

**Pull request merged**